tests/integration/test_sqs.py

```diff
@@ -1398,16 +1400,631 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
+        i = 0
+        while len(result_recv["Messages"]) < message_count and i < 3:
+            result_recv = sqs_client.receive_message(
+                QueueUrl=queue_url, MaxNumberOfMessages=message_count, VisibilityTimeout=0
+            )
+            i += 1
+        messages_recv = result_recv["Messages"]
+        assert len(messages_recv) == message_count
+
+        ids_sent = []
+        ids_received = []
+        for i in range(message_count):
+            ids_sent.append(successful[i]["MessageId"])
+            ids_received.append((messages_recv[i]["MessageId"]))
+
+        assert set(ids_sent) == set(ids_received)
+
+        delete_entries = [
+            {"Id": message["MessageId"], "ReceiptHandle": message["ReceiptHandle"]}
+            for message in messages_recv
+        ]
+        sqs_client.delete_message_batch(QueueUrl=queue_url, Entries=delete_entries)
+        confirmation = sqs_client.receive_message(
+            QueueUrl=queue_url, MaxNumberOfMessages=message_count
+        )
+        assert "Messages" not in confirmation.keys()
+
+    def test_create_and_send_to_fifo_queue(self, sqs_client, sqs_create_queue):
+        # Old name: test_create_fifo_queue
+        queue_name = "queue-{}.fifo".format(short_uid())
+        attributes = {"FifoQueue": "true"}
+        queue_url = sqs_create_queue(QueueName=queue_name, Attributes=attributes)
+
+        # it should preserve .fifo in the queue name
+        assert queue_name in queue_url
+
+        message_id = sqs_client.send_message(
+            QueueUrl=queue_url,
+            MessageBody="test",
+            MessageDeduplicationId="dedup-{}".format(short_uid()),
+            MessageGroupId="test_group",
+        )["MessageId"]
+
+        result_recv = sqs_client.receive_message(QueueUrl=queue_url)
+        assert result_recv["Messages"][0]["MessageId"] == message_id
+
+    def test_fifo_queue_requires_suffix(self, sqs_create_queue):
+        queue_name = "invalid-{}".format(short_uid())
+        attributes = {"FifoQueue": "true"}
+
+        with pytest.raises(Exception) as e:
+            sqs_create_queue(QueueName=queue_name, Attributes=attributes)
+        e.match("InvalidParameterValue")
+
+    def test_set_queue_policy(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        attributes = {"Policy": TEST_POLICY}
+        sqs_client.set_queue_attributes(QueueUrl=queue_url, Attributes=attributes)
+
+        # accessing the policy generally and specifically
+        attributes = sqs_client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=["All"])[
+            "Attributes"
+        ]
+        assert "sqs:SendMessage" in attributes["Policy"]
+        attributes = sqs_client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=["Policy"])[
+            "Attributes"
+        ]
+        assert "sqs:SendMessage" in attributes["Policy"]
```

Comment:
could we also test whether the policy attributes are set correctly?

Reply:
Done